scipy/sparse/linalg/eigen/arpack/arpack.py

```diff
@@ -1231,8 +1232,26 @@ def eigs(A, k=6, M=None, sigma=None, which='LM', v0=None,
             import warnings
             warnings.warn('M does not have the same type precision as A. '
                           'This may adversely affect ARPACK convergence')
+
     n = A.shape[0]
 
+    if k >= n - 1:
+        import warnings
+        warnings.warn('k greater than/equal to N - 1 for N * N square matrix. '
+                      'Using scipy.linalg.eig instead.',
+                      SparseEfficiencyWarning)
+
+        if isinstance(A, LinearOperator):
+            raise ValueError("'A' cannot be a LinearOperator since"
+                             "scipy.linalg.eig is being used.")
+        if isinstance(M, LinearOperator):
+            raise ValueError("'M' cannot be a LinearOperator since"
+                             "scipy.linalg.eig is being used.")
+        if not isdense(A):
+            A = A.todense()
```

Comment:
No, we still need to support all three input types. I'm suggesting that be behavior for non-ndarray types (spmatrix, LinearOperator) remain as it was before this PR, but with an informative error message so that the user knows how to proceed.

Reply:
Okay. I get it.